addons/mail/tests/test_mail_channel.py

```diff
@@ -24,128 +24,216 @@ def setUpClass(cls):
+            self.env['mail.channel'].browse(self.test_group.id).unlink()
         with self.assertRaises(AccessError):
-            self.env['mail.channel'].browse(self.group_public.id).unlink()
+            self.env['mail.channel'].browse(self.test_chat.id).unlink()
 
     @mute_logger('odoo.addons.base.models.ir_rule', 'odoo.models', 'odoo.models.unlink')
     @users('employee')
     def test_access_employee(self):
-        # Employee read employee-based group: ok
-        group_groups = self.env['mail.channel'].browse(self.group_groups.id)
-        group_groups.read()
-
-        # Employee can create a group
-        new_channel = self.env['mail.channel'].create({'name': 'Test'})
-        self.assertIn(new_channel.channel_partner_ids, self.partner_employee)
-
-        # Employee update employee-based group: ok
-        group_groups.write({'name': 'modified'})
-
-        # Employee unlink employee-based group: ko
+        # Read channel -> ok
+        self.env['mail.channel'].browse(self.test_channel.id).read()
+        # Read group -> ko, restricted to members
         with self.assertRaises(AccessError):
-            group_groups.unlink()
-
-        # Employee cannot read a private group
+            self.env['mail.channel'].browse(self.test_group.id).read()
+        # Read chat -> ko, restricted to members
         with self.assertRaises(AccessError):
-            self.env['mail.channel'].browse(self.group_private.id).read()
+            self.env['mail.channel'].browse(self.test_chat.id).read()
 
-        # Employee cannot write on private
+        # Update group/chat when not being a member: ko, no access rights
+        with self.assertRaises(AccessError):
+            self.env['mail.channel'].browse(
+                self.test_group.id).write({'name': 'modified'})
         with self.assertRaises(AccessError):
-            self.env['mail.channel'].browse(self.group_private.id).write({'name': 're-modified'})
+            self.env['mail.channel'].browse(
+                self.test_chat.id).write({'name': 'modified'})
+
+        # Being a channel/group/chat member: -> ok
+        self.test_channel.sudo().add_members(self.env.user.partner_id.id)
+        self.test_group.sudo().add_members(self.env.user.partner_id.id)
+        self.test_chat.sudo().add_members(self.env.user.partner_id.id)
+
+        # Read a group when being a member: ok
+        self.env['mail.channel'].browse(self.test_group.id).read()
+        # Read a chat when being a member: ok
+        self.env['mail.channel'].browse(self.test_chat.id).read()
+
+        # Update channel/group/chat when being a member: ok
+        self.env['mail.channel'].browse(
+            self.test_channel.id).write({'name': 'modified again'})
+        self.env['mail.channel'].browse(
+            self.test_group.id).write({'name': 'modified again'})
+        self.env['mail.channel'].browse(
+            self.test_chat.id).write({'name': 'modified again'})
```

Comment:
All changes in test formats should be done in a prep PR (that doesn't change business code at the same time), so that we can see what actually changes (if anything) with this one only in terms of assertions.